ACPs/77-reinventing-subnets/README.md

```diff
@@ -132,37 +209,55 @@ Since there are no `EndTime`s enforced by the P-Chain, Subnets must use this tra
     //   - Nonce
-    //   - Weight of the validator
-    //   - BLS multisig over the above payload
+    //   - Weight
     Message warp.Message `json:"message"`
 }
 ```
 
-The `Message` field in the above transaction must be an Avalanche Warp Message using the `AddressedCall` format with the below payload. A `typeID` for distinguishing between the different payload types on the P-Chain must be included prior to this ACP being considered "Implementable".
+The `Message` field must be an `AddressedCall` with the payload:
 
 ```text
++--------------+----------+----------+
+|      codecID :   uint16 |  2 bytes |
++--------------+----------+----------+
+|       typeID :   uint32 |  4 bytes |
 +-----------+----------+----------+
-| messageID : [32]byte | 32 bytes |
-+-----------+----------+----------+
-|     nonce :   uint64 |  8 bytes |
-+-----------+----------+----------+
-|    weight :   uint64 |  8 bytes |
-+-----------+----------+----------+
-                       | 48 bytes |
-                       +----------+
+| validationID : [32]byte | 32 bytes |
++--------------+----------+----------+
+|        nonce :   uint64 |  8 bytes |
++--------------+----------+----------+
+|       weight :   uint64 |  8 bytes |
++--------------+----------+----------+
+                          | 54 bytes |
+                          +----------+
 ```
 
-- `messageID` is the sha256 hash of the Warp message in the `RegisterSubnetValidatorTx` that registered the Subnet Validator
-- `nonce` is a strictly increasing number that denotes the latest validator weight update and provides replay protection for this transaction. The P-Chain state will store a `minNonce` associated with `messageID`. When accepting the `RegisterSubnetValidatorTx`, the `minNonce` was set to `0` for `messageID`. `nonce` must satisfy `nonce >= minNonce` for `SetSubnetValidatorWeightTx` to be valid. Note that `nonce` is not required to be incremented by `1` with each successive validator weight update. If `minNonce` is `MaxUint64`, the `weight` in the `SetSubnetValidatorWeightTx` is required to be `0` to prevent Subnets from being unable to remove `nodeID` in a future transaction. When a Subnet Validator is removed from the active set (`weight == 0`), the `minNonce` and `messageID` will be removed from the P-Chain state. This state can be reaped during validator exit since `messageID` can never be re-initialized as a result of the replay protection provided by `expiry` in `RegisterSubnetValidatorTx`. `minNonce` will be set to `nonce + 1` when `SetSubnetValidatorWeightTx` is executed and `weight != 0`.
+- `codecID` is the codec version used to serialize the payload and is hardcoded to `0x0000`
+- `typeID` is the payload type identifier and is `0x00000001` for this transaction
+- `validationID` is the SHA256 of the `Payload` of the `AddressedCall` in the `RegisterSubnetValidatorTx` adding the validator to the Subnet's validator set
```

Comment:
Are the `codecID` and `typeID` included in the `SHA256`?

Reply:
YEs